content/en/docs/features/monitoring-and-diagnostics/monitoring-device-connectivity.md

```diff
@@ -0,0 +1,32 @@
+---
+title: 'Monitoring Device Connectivity'
+description: 'Learn how to determine the online or offline status of connected devices'
+date: '2023-08-08'
+categories: [features,iot-connectivity]
+keywords: [service, coap-gateway, network-monitoring]
+weight: 2
+---
+
+The CoAP gateway is responsible for the connection management including the up to date representation of the device status. This information is driven through the Resource Aggregate command - `UpdateDeviceMetadata`, that gets initiated when device connection status changes - device connects or disconnects from the CoAP Gateway. However, there are various scenarios to consider, such as unexpected termination of the CoAP Gateway, due to factors like node failure, power loss or others.
+
+## Managing Service Status Updates
+
+Every instance of the CoAP gateway is assigned a unique ID during initialization. These IDs are not only stored in a database but are also linked to device metadata to facilitate online state updates.
+
+At regular intervals, each CoAP gateway instance collaboratively updates a shared record containing their respective information, primarily their unique IDs. This update takes place periodically, typically once every minute. The specific record being updated changes over time, following a cyclic pattern (e.g., within a 1-minute period, each update corresponds to a different record). In the database, both the current shared record and the preceding one are retained.
```

Comment:
In case of 10k devices, are you updating 10k documents with bulk write? What is the performance impact? Did you measure it what's the impact if you do it e.g. every 5 seconds?

Reply:
It's not clear what do you really store per each device. Once you talk about the unique ID you store during initialisation in each device metadata of each device when it comes online, and then you talk about collaborative update of a shared record.
1) what is a shared record
2) how is it shared - what does it mean that it's shared
3) where is it persisted
4) what exactly "collaboratively updates" means. Who collaborate on this update? Multiple CoAP Gateways at once?

Reply:
For each coap-gateway you will update the shared resource. If you have 10k devices and 5 coap-gateways then it is only 5 updates per minute.

 During updating device metadata you only safe coap-gateway-id during the set online state.